### PR TITLE
Add clear uma auth state button to react example

### PR DESCRIPTION
**Pull request by bsiaotickchong**

### CODE REVIEW

**Comment by bsiaotickchong** [comment]
* **#150** <a href="https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/150?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#148** <a href="https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/148?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bsiaotickchong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/uma-auth-client/src/hooks/useOAuth.tsx [hunk 1]
```diff
@@ -104,6 +105,16 @@ export const useOAuth = create<OAuthState>()(
...
       },
+      clearUserAuth: () => {
+        set({
+          token: undefined,
+          nwcConnectionUri: undefined,
+          nwcExpiresAt: undefined,
+          codeVerifier: undefined,
+          csrfState: undefined,
+          uma: undefined,
+        });
```
Comment (jklein24) [inline packages/uma-auth-client/src/hooks/useOAuth.tsx:116]: Could this just be `set({})` or did ts not like that for some reason?

**Review by jklein24 — APPROVED** [review]

File: packages/uma-auth-client/src/hooks/useOAuth.tsx (hunk 1 above)
Comment (bsiaotickchong) [reply packages/uma-auth-client/src/hooks/useOAuth.tsx:116]: If the property isn't specified zustand doesn't update the state for it, so in this case we need to manually specify undefined

we also need to keep the authConfig, so this isn't actually clearing everything

**Comment by bsiaotickchong** [comment]
### Merge activity

* **Sep 16, 2:13 PM EDT**: @bsiaotickchong started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/150).
* **Sep 16, 2:15 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/150) rebased this pull request as part of a merge.
* **Sep 16, 2:16 PM EDT**: @bsiaotickchong merged this pull request with [Graphite](https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/150).

**Pull request merged**